crates/network/src/lib.rs

```diff
@@ -0,0 +1,386 @@
+    )>,
+) {
+    let (mut connections, mut packet_events, mut disconnect_events, mut commands) =
+        state.get_mut(world);
+
+    for (entity, mut connection) in &mut connections {
+        let result = connection.try_recv();
+
+        match result {
+            Ok(frame) => {
+                packet_events.send(PacketReceivedEvent {
+                    connection: entity,
+                    frame,
+                });
+            }
+            Err(TryRecvError::Empty) => {}
+            Err(TryRecvError::Disconnected) => {
+                tracing::info!("Client disconnected");
+                disconnect_events.send(DisconnectEvent { connection: entity });
+                commands.entity(entity).remove::<RemoteConnection>();
+            }
+        }
+    }
+
+    state.apply(world);
+}
+
+fn connection_event_handler(
+    mut packets: EventReader<PacketReceivedEvent>,
+    mut handshake_events: EventWriter<HandshakeEvent>,
+    mut client_status_request_events: EventWriter<ClientStatusRequestEvent>,
+    mut connections: Query<&mut RemoteConnection>,
+) {
+    for packet in packets.iter() {
+        let payload = &packet.frame.payload;
+        if let Ok(mut connection) = connections.get_mut(packet.connection) {
+            match &connection.state {
+                HandshakeState::Handshaking => {
+                    match ClientHandshakePacket::decode(&mut std::io::Cursor::new(payload)) {
+                        Ok(p) => match p {
+                            ClientHandshakePacket::Handshake(p) => {
+                                handshake_events.send(HandshakeEvent {
+                                    connection: packet.connection,
+                                    protocol_version: p.protocol_version,
+                                    server_address: p.server_address,
+                                    server_port: p.server_port,
+                                    next_state: p.next_state,
+                                });
+                                connection.state = p.next_state;
+                            }
+                        },
+                        Err(e) => {
+                            tracing::error!("error: {:?}", e)
+                        }
+                    };
+                }
+                HandshakeState::Status => {
+                    match ClientStatusPacket::decode(&mut std::io::Cursor::new(payload)) {
+                        Ok(p) => match p {
+                            ClientStatusPacket::Request(_) => {
+                                client_status_request_events.send(ClientStatusRequestEvent {
+                                    connection: packet.connection,
+                                })
+                            }
+                            ClientStatusPacket::Ping(ping) => {
+                                connection.send_packet(
+                                        rjacraft_protocol::packets::server::ServerStatusPacket::Pong(
```

Comment:
Definitely a case for an import (especially, considering that `ServerStatusPacket` is unlikely to collide with anything else due to `Packet`  suffix).